http_parser.cc

```diff
@@ -0,0 +1,104 @@
+#include "http_parser.h"
+#include <string>
+#include <vector>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/regex.hpp>
+
+HttpParser* HttpParser::MakeHttpParser(const char* const raw_req)
```

Comment:
In connection.cc, data has a max length of 1024. So no matter what, it'll have a \0, but yes, we are limited to a 1024 byte request right now....